model/map/src/main/java/org/keycloak/models/map/storage/chm/ConcurrentHashMapStorage.java

```diff
@@ -159,16 +154,11 @@ public long delete(QueryParameters<M> queryParameters) {
         return res.get();
     }
 
-    @Override
-    public ModelCriteriaBuilder<M> createCriteriaBuilder() {
-        return new MapModelCriteriaBuilder<>(keyConvertor, fieldPredicates);
-    }
```

Comment:
Could this method be kept here (without `@Override`) and used from within the transaction?